src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass_LocalFunctions.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal partial class AbstractFlowPass<TLocalState, TLocalFunctionState>
+    {
+        internal abstract class AbstractLocalFunctionState
+        {
+            /// <summary>
+            /// This is the part of the local function transfer function which moves the lattice
+            /// upward. It starts unreachable, as the local function is not assumed to be called
+            /// until proven otherwise. When a local function is called, this state is <see
+            /// cref="Meet(ref TLocalState, ref TLocalState)"/> with the current state.
+            /// </summary>
```

Comment:
I don't think this is wrong so much as not helpful.  I think it would be helpful to go into more detail about the whole strategy.  This is a bit confusing, for example because `Meet` moves down the lattice.